lib/logflare_web/controllers/plugs/verify_api_access.ex

```diff
@@ -4,43 +4,70 @@ defmodule LogflareWeb.Plugs.VerifyApiAccess do
 
   Assigns the token's associated user if the token is provided
 
-  Options:
-  - `:resource`, required - an atom determining the resource accessed, as each resource has its own authorization verification flow.
+  Authentication api key can either be through access tokens or legacy `user.api_key`.
+
+  Access token usage is preferred and `user.api_key` is only used as a fallback.
   """
   import Plug.Conn
   import Phoenix.Controller
-  alias Logflare.{Auth, Endpoints}
+  alias Logflare.Auth
+  alias Logflare.Endpoints
+  alias Logflare.Users
 
   def init(args), do: args |> Enum.into(%{})
 
-  def call(conn, opts) do
-    opts
-    |> Map.get(:resource)
+  def call(conn, _opts) do
+    conn.request_path
+    |> case do
+      "/api/endpoints/" <> _ -> :endpoints
+      "/endpoints" <> _ -> :endpoints
+      _ -> :generic
+    end
     |> do_auth(conn)
   end
 
-  defp do_auth(:endpoints, conn) do
+  defp do_auth(:endpoints, %{params: params} = conn) when is_map_key(params, "token") do
     conn = fetch_query_params(conn)
     # fetch endpoint info
-    with endpoint_token <- conn.params["token"],
-         %Endpoints.Query{enable_auth: true, user_id: user_id} <-
-           Endpoints.get_query_by_token(endpoint_token),
-         {:ok, token} <- extract_token(conn),
-         {:ok, user} <- Auth.verify_access_token(token),
-         true <- user_id == user.id do
-      assign(conn, :user, user)
+    endpoint = Endpoints.get_query_by_token(conn.params["token"])
```

Comment:
We should set the mapped query in the assigns once in the plug pipeline (probably in a different plug) and we'll have it in the controller. 

We're currently doing the db lookup twice because of this and then where we do `get_mapped_query_by_token` in the EndpointController.

Reply:
👌 can do this in a resource fetching plug.